### PR TITLE
Add non-rated issues to Finish page

### PR DESCRIPTION
**Pull request by leikkisa**
Connects #5501

### CODE REVIEW

File: app/assets/stylesheets/_intake.scss [hunk 1]
```diff
@@ -31,14 +31,38 @@
...
+  }
+
+  label {
+    margin-bottom: 8px;
+  }
+
+  > span {
+    align-self: end;
+    padding-bottom: 5px;
+  }
```
Comment (shanear) [inline app/assets/stylesheets/_intake.scss]: I think we should really try and use style guide stuff as much as possible. Can we add a tech improvement story to see what we can extract into the style guide from this?

File: client/app/intake/components/NonRatedIssue.jsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  handleCategoryChange(event) {
+    this.props.setIssueCategory(this.props.issueId, event.value);
+  }
+
+  handleDescriptionChange(event) {
+    this.props.setIssueDescription(this.props.issueId, event);
+  }
+
+  render () {
+    const { nonRatedIssues } = this.props;
```
Comment (shanear) [inline client/app/intake/components/NonRatedIssue.jsx]: We should be passing just the NonRated issue associated with this component here. Not the entire list

File: client/app/intake/constants.js [hunk 3]
```diff
@@ -133,3 +136,40 @@ export const CANCELLATION_REASONS = {
...
+{ value: 'Dependency Spouse - Deemed valid marriage',
+  label: 'Dependency Spouse - Deemed valid marriage' },
+{ value: 'Military Retired Pay',
+  label: 'Military Retired Pay' },
+{ value: 'Contested Claims (other than apportionment)',
+  label: 'Contested Claims (other than apportionment)' },
+{ value: 'Lack of Qualifying Service',
+  label: 'Lack of Qualifying Service' },
+{ value: 'Other non-rated',
+  label: 'Other non-rated' }];
```
Comment (shanear) [inline client/app/intake/constants.js]: If all these are the exact same `value` and `label`, we should be able to dynamically generate this structure without having to copy and paste each value twice.

File: client/app/intake/pages/appeal/nonRatedIssues.jsx [hunk 4]
```diff
@@ -1,18 +1,55 @@
...
-    <p>
+class NonRatedIssuesUnconnected extends React.PureComponent {
+  render() {
+    const { nonRatedIssues } = this.props;
+
+    const nonRatedIssuesSection = _.map(nonRatedIssues, (issue, issueId) => {
+      return (
+        <NonRatedIssue key={issueId} issueId={issueId} />
+      );
+    });
```
Comment (shanear) [inline client/app/intake/pages/appeal/nonRatedIssues.jsx]: It's probably best to pass the nonRatedIssue as a whole into the component instead of the id.

File: client/app/intake/pages/supplementalClaim/nonRatedIssues.jsx [hunk 5]
```diff
@@ -1,18 +1,55 @@
...
+    setIssueDescription
+  }, dispatch)
+)(NonRatedIssueUnconnected);
+
+const AddIssueButton = connect(
+  ({ supplementalClaim }) => ({ nonRatedIssues: supplementalClaim.nonRatedIssues }),
+  (dispatch) => bindActionCreators({
+    addNonRatedIssue
+  }, dispatch)
+)(AddIssueButtonUnconnected);
```
Comment (shanear) [inline client/app/intake/pages/supplementalClaim/nonRatedIssues.jsx]: Woof, yeah there is a lot of duplication between these 3. We can save that for our refactoring ticket.

File: client/app/intake/reducers/appeal.js [hunk 6]
```diff
@@ -172,6 +176,39 @@ export const appealReducer = (state = mapDataToInitialAppeal(), action) => {
...
+        }
+      }
+    });
+  case ACTIONS.SET_ISSUE_CATEGORY:
+    return update(state, {
+      nonRatedIssues: {
+        [action.payload.issueId]: {
+          $set: {
+            issueCategory: action.payload.category,
+            issueDescription: state.nonRatedIssues[action.payload.issueId].issueDescription
```
Comment (shanear) [inline client/app/intake/reducers/appeal.js]: You shouldn't need this line to set it to what it was

File: client/app/intake/reducers/appeal.js [hunk 7]
```diff
@@ -172,6 +176,39 @@ export const appealReducer = (state = mapDataToInitialAppeal(), action) => {
...
+          }
+        }
+      }
+    });
+  case ACTIONS.SET_ISSUE_DESCRIPTION:
+    return update(state, {
+      nonRatedIssues: {
+        [action.payload.issueId]: {
+          $set: {
+            issueCategory: state.nonRatedIssues[action.payload.issueId].issueCategory,
```
Comment (shanear) [inline client/app/intake/reducers/appeal.js]: ditto from above.

File: client/app/intake/util/index.js [hunk 8]
```diff
@@ -34,3 +34,35 @@ export const formatRatings = (ratings) => {
...
+          });
+        }).
+        flatten().
+        filter('isSelected')
+  };
+
+  const nonRatingData = {
+    request_issues:
+      _.map(intakeState.nonRatedIssues, (issue) => {
+        if (issue.issueCategory) {
```
Comment (shanear) [inline client/app/intake/util/index.js]: I think this returns nil for issues without a category. We want to filter them out instead.

File: client/app/intake/reducers/appeal.js [hunk 9]
```diff
@@ -32,6 +32,9 @@ const updateFromServerIntake = (state, serverIntake) => {
     selectedRatingCount: {
       $set: state.selectedRatingCount
     },
+    nonRatedIssues: {
+      $set: state.nonRatedIssues
```
Comment (shanear) [inline client/app/intake/reducers/appeal.js]: where is this coming from? Why is this needed?

File: app/assets/stylesheets/_intake.scss (hunk 1 above)
Comment (leikkisa) [reply app/assets/stylesheets/_intake.scss]: Oh, this was for the "Save" button so I can remove this specific section.

File: client/app/intake/components/NonRatedIssue.jsx (hunk 2 above)
Comment (leikkisa) [reply client/app/intake/components/NonRatedIssue.jsx]: I think I'll need you to show me how to do this

File: client/app/intake/constants.js (hunk 3 above)
Comment (leikkisa) [reply client/app/intake/constants.js]: Done. Seems less efficient to me, but it is prettier code

File: client/app/intake/reducers/appeal.js (hunk 6 above)
Comment (leikkisa) [reply client/app/intake/reducers/appeal.js]: I haven't been able to get it to work without it, when it updates one it deletes the other. Would be happy to work on this with help.

File: client/app/intake/reducers/appeal.js (hunk 9 above)
Comment (leikkisa) [reply client/app/intake/reducers/appeal.js]: You're right! Removed. I noticed also when starting to work on the Different Claimant ticket

File: client/app/intake/pages/supplementalClaim/nonRatedIssues.jsx (hunk 5 above)
Comment (leikkisa) [reply client/app/intake/pages/supplementalClaim/nonRatedIssues.jsx]: Cool, would love help with that. I tried to make as much of the "unconnected" pieces in a reusable component, but the entire issues list has the Add Issue button in it, and refactoring that out messed it up, so I left that in each individual flow.

File: app/assets/stylesheets/_intake.scss (hunk 1 above)
Comment (leikkisa) [reply app/assets/stylesheets/_intake.scss]: I made a ticket for this in Tech Backlog

File: client/app/intake/util/index.js (hunk 8 above)
Comment (leikkisa) [reply client/app/intake/util/index.js]: Done

File: client/app/intake/reducers/appeal.js (hunk 6 above)
Comment (leikkisa) [reply client/app/intake/reducers/appeal.js]: Done

File: client/app/intake/reducers/appeal.js (hunk 7 above)
Comment (leikkisa) [reply client/app/intake/reducers/appeal.js]: Done

File: client/app/intake/pages/appeal/nonRatedIssues.jsx (hunk 4 above)
Comment (leikkisa) [reply client/app/intake/pages/appeal/nonRatedIssues.jsx]: Done

File: client/app/intake/components/NonRatedIssue.jsx (hunk 2 above)
Comment (leikkisa) [reply client/app/intake/components/NonRatedIssue.jsx]: Done
